tasks/helpers/getPackagePaths.js

```diff
@@ -41,23 +37,36 @@ var recursivelyAccumulateRequiredPaths = function(accumPaths, hostPath) {
     // Add the paths to our list and recursively search their associated files.
     .reduce(function(accumPaths, relativeRequiredPath) {
       var normalizedRequiredPath = path.join(path.dirname(hostPath), relativeRequiredPath);
-      return recursivelyAccumulateRequiredPaths(accumPaths, normalizedRequiredPath);
+
+      // only process the require statements for a file we haven't seen yet to avoid circular imports
```

Comment:
In the `R.compose` function inside the `getLibPaths` function, we were filtering out (potentially) many instances of the same occurred file. consider:

```
constants.js
someFile.js
someFile2.js
```

if both `someFile.js` and `someFile2.js` both import `constants.js`, then `constants.js` exists in this recursively accumulated array twice.

When `someFile.js` calls `require(someFile2)` and `someFile2` calls `require(someFile)`, then this function keeps constantly bouncing back and forth and creating new recursive frames as we pull off every `matchRequries` result over and over.